mrmustard/lab_dev/circuit_components.py

```diff
@@ -602,9 +609,39 @@ def to_bargmann(self) -> CircuitComponent:
             >>> assert d_bargmann.wires == d.wires
             >>> assert isinstance(d_bargmann.representation, Bargmann)
         """
-        if isinstance(self.representation, Bargmann):
+        from .circuit_components_utils import BtoQ, BtoPS
+        if isinstance(self.representation, Bargmann):# TODO: better name for Bargmann class
+            # check cc rep
+            if not indices:
+                indices = self.wires.indices
+            
+            for i in indices:
+                name, arg =  self._index_representation[i]
+                if name == 'Q':
+                    self._index_representation[i] = ('B', None)
+                    if i in self.wires.output.bra.indices:
+                        self = self @ BtoQ([self.wires.index_to_mode_dict[i]], phi = arg).adjoint.inverse()
+                    if i in self.wires.output.ket.indices:
+                        self = self @ BtoQ([self.wires.index_to_mode_dict[i]], phi = arg).inverse()
+                    if i in self.wires.input.bra.indices:
+                        self = BtoQ([self.wires.index_to_mode_dict[i]], phi = arg).dual.adjoint.inverse() @ self
+                    if i in self.wires.input.bra.indices:
+                        self = BtoQ([self.wires.index_to_mode_dict[i]], phi = arg).dual.inverse() @ self 
+                    
+
+                if name == 'PS':
+                    self._index_representation[i] = ('B', None)
+                    if i in self.wires.output.bra.indices:
+                        self = self @ BtoPS([self.wires.index_to_mode_dict[i]], s = arg).adjoint.inverse()
+                    if i in self.wires.output.ket.indices:
+                        self = self @ BtoPS([self.wires.index_to_mode_dict[i]], s = arg).inverse()
+                    if i in self.wires.input.bra.indices:
+                        self = BtoPS([self.wires.index_to_mode_dict[i]], s = arg).dual.adjoint.inverse() @ self
+                    if i in self.wires.input.bra.indices:
+                        self = BtoPS([self.wires.index_to_mode_dict[i]], s = arg).dual.inverse() @ self 
+
             return self
-        else:
```

Comment:
Any reason for this? I believe else is better (as long as there are no other representation cases) because it avoids an isinstance call